packages/plugin-ext/src/plugin/node/debug/debug.ts

```diff
@@ -180,6 +183,31 @@ export class DebugExtImpl implements DebugExt {
         return Disposable.create(() => this.descriptorFactories.delete(debugType));
     }
 
+    static SCHEME = 'debug';
+    static SCHEME_PATTERN = /^[a-zA-Z][a-zA-Z0-9\+\-\.]+:/;
+    asDebugSourceUri(source: theia.DebugProtocolSource, session?: theia.DebugSession): theia.Uri {
+        const raw = source as DebugProtocol.Source;
+        const uri = this.asDebugSourceURI(raw, session?.id);
+        return URIImpl.parse(uri.toString());
+    }
+
+    private asDebugSourceURI(raw: DebugProtocol.Source, sessionId?: string): TheiaURI {
+        if (raw.sourceReference && raw.sourceReference > 0) {
+            let query = String(raw.sourceReference);
```

Comment:
Since this is only a number, it doesn't appear to be correctly formatted for a query string?